From 1570f844c9385a0e84106cc6535b4b8c1bd19f85 Mon Sep 17 00:00:00 2001
From: NeilBrown <neilb@suse.de>
Date: Thu, 10 Feb 2022 08:57:35 +1100
Subject: [PATCH] Test TCP request correctly in nfs_get_info()

The TCP_REQUESTED flag is masked out by the caller, so it never gets to
nfs_get_info().
We can test if TCP was requested by examining the 'proto' parameter.

Signed-off-by: NeilBrown <neilb@suse.de>
---
 modules/replicated.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/modules/replicated.c
+++ b/modules/replicated.c
@@ -291,7 +291,7 @@ static unsigned int get_nfs_info(unsigne
 
 	rpc_info->proto = proto;
 	if (port < 0) {
-		if ((version & NFS4_REQUESTED) && (version & TCP_REQUESTED))
+		if ((version & NFS4_REQUESTED) && (proto == IPPROTO_TCP))
 			rpc_info->port = NFS_PORT;
 		else
 			port = 0;
